primitives/account/src/account/staking_contract/traits.rs

```diff
@@ -604,13 +640,17 @@ impl AccountTransactionInteraction for StakingContract {
                 // Get the staker address from the proof.
                 let staker_address = proof.compute_signer();
 
+                // Fetch the staker.
                 let staker = store.expect_staker(&staker_address)?;
 
-                self.can_remove_stake(&store, &staker, block_state.number)?;
+                // Verify that the stake can actually be removed.
+                self.can_remove_stake(&staker, transaction.total_value())?;
+                // Verify that the fee by itself can be removed without violating the minimum stake.
+                self.can_remove_stake(&staker, transaction.fee)?;
```

Comment:
Yes, correct. That was a trade-off we agreed upon to actually enforce the minimum stake despite the possibility of a commit failed tx and its consequential fee deduction.
Since it's an edge case that only happens if a staker is only holding the minimum stake (which is a really small amount) and fees are currently 0 on our network, we decided to enforce the minimum stake at the expense of this edge case.
Otherwise, we would have a minimum stake but it could be broken, similar to the validator deposit. Calling it a minimum stake would be rather pointless, even implementing a "minimum stake" on these conditions would be rather pointless.

Reply:
I don't fully understand. Is this code here the ultimate test if the transaction will succeed? Or can it still fail later on?

If it can still fail later on, I agree that this test has to be as it is.

But if this is the ultimate test, and we have already established that the whole tx will go through with the first check (e.g. removing 99 NIM with 1 NIM fee), then why do we still need to test for a failing tx, which is the second check?

Reply:
We cannot be sure that the tx will be successful. It can fail on the receiver but trying to transfer to another contract for example. That's just an example, there's also some other scenarios where it can still fail later at the commit failed tx.

Reply:
Thanks for clarifying :bulb: